spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/rdd/RDDSpec.scala

```diff
@@ -201,6 +201,38 @@ class RDDSpec extends SparkCassandraITFlatSpecBase {
     checkArrayCassandraRow(result)
   }
 
+  it should "throw a meaningful exception if partition column is null when joining with Cassandra table" in {
+    val source = sc.parallelize(keys).map(x ⇒ new KVWithOptionRow(None))
+    try {
+      source.joinWithCassandraTable[(Int, Long, String)](ks, tableName).collect()
+      fail()
+    } catch {
+      case ex: Exception ⇒
+        ex.getMessage should include("Invalid null value for partition key part key")
```

Comment:
This way you're not testing if the exception has been thrown at all. Better use the pattern recommended by scalatest: 
val e = the [Exception] thrown by { some code }
e.getMessage should ...